packages/graphback-datasync/src/providers/DeltaMongoDBDataProvider.ts

```diff
@@ -0,0 +1,121 @@
+import { GraphQLObjectType } from 'graphql';
+import { getDatabaseArguments, metadataMap, GraphbackContext, NoDataError, TransformType, FieldTransform, GraphbackOrderBy, GraphbackPage, QueryFilter, StringInput, GraphbackOperationType } from '@graphback/core';
+import { ObjectId } from 'mongodb';
+import { MongoDBDataProvider, applyIndexes } from '@graphback/runtime-mongo';
+import { ConflictError, ConflictStateMap, ConflictEngine, HashConflictEngine } from "../util";
+import { DataSyncProvider, DataSyncMongoDBDataProvider, DeltaProvider } from "./";
+
+/**
+ * Mongo provider that attains data synchronization using soft deletes
+ */
+export class DeltaMongoDBDataProvider<Type = any> extends DataSyncMongoDBDataProvider <Type> {
```

Comment:
I will need to have map/short doc of providers we add and how they relay with each other. We can see if we can simplify things. 

For me DataSync is always Delta. Conflict can be optional but we can also leverage this - maybe we should always have versioned conflict enabled etc. - all depends on how complex conflict check will become etc.

Reply:
I am a bit lost here: Do we want opt-in conflict or conflict all the time?

Reply:
See explanations above.